pil-analyzer/src/evaluator.rs

```diff
@@ -1818,4 +1813,31 @@ mod test {
 
         assert_eq!(parse_and_evaluate_symbol(input, "F::r"), "5".to_string());
     }
+
+    #[test]
+    fn traits_generic_function() {
+        let input = "
+        namespace std::convert(4);
+            let fe = || fe();
+        namespace F(4);
+            trait Add<T> {
+                add: T, T -> T,
+            }
+            impl Add<int> {
+                add: |a, b| a + b,
+            }
+            impl Add<fe> {
+                add: |a, b| a + b,
```

Comment:
Can you maybe use `a + 2*b` or something? The evaluator does not take the determined types into account, but only the runtime types - so even if we pick the wrong `add` function here in the evaluator, the result would still be correct.